### ISSUE TITLE
Add loading icon to Registration screen

### ISSUE THREAD

**Issue by theRoughCode** [theRoughCode/WatsMyMajor#64]
When a user submits the registration form, we want there to be some sort of indication that we are processing the request.  There can be a loading icon in place of the "Sign Up" text in the submit button.

**Comment by mrassili** [comment]
Hi, I'm Marouane and I'm really keen to help out implement this. Could you please assign it to me ? Thanks

**Comment by mrassili** [comment]
After running `npm install` with some warnings but no errors, when I run `npm start` I get this error : 

```
Marouanes-MBP:WatsMyMajorBeta mrassili$ npm start

> WatsMyMajor@1.0.0 start /Users/mrassili/projects/WatsMyMajorBeta
> node server

undefined:1
undefined
^

SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at Object.<anonymous> (/Users/mrassili/projects/WatsMyMajorBeta/server/models/database/index.js:6:29)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/mrassili/projects/WatsMyMajorBeta/server/models/database/users.js:3:22)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/mrassili/projects/WatsMyMajorBeta/server/routes/auth.js:4:15)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/mrassili/projects/WatsMyMajorBeta/server/routes/index.js:17:21)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/mrassili/projects/WatsMyMajorBeta/server/router.js:6:23)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/Users/mrassili/projects/WatsMyMajorBeta/server/index.js:7:16)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! WatsMyMajor@1.0.0 start: `node server`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the WatsMyMajor@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mrassili/.npm/_logs/2018-08-25T23_04_37_223Z-debug.log
```

What could be the problem ? I have Node v8.11.3 & npm v5.6.0

**Comment by Jhong098** [comment]
I would like to work on this

**Comment by theRoughCode** [comment]
Fixed by PR #129
